sources/packages/web/src/components/aest/students/bypassedRestrictions/HistoryBypassedRestrictions.vue

```diff
@@ -0,0 +1,127 @@
+<template>
+  <v-card>
+    <v-container>
+      <body-header title="History of Bypassed Restrictions" class="m-1">
+        <template #actions>
+          <check-permission-role :role="Role.StudentAddRestriction">
+            <template #="{ notAllowed }">
+              <v-btn
+                @click="addBypassedRestriction"
+                class="float-right"
+                color="primary"
+                data-cy="addRestrictionButton"
+                prepend-icon="fa:fa fa-plus-circle"
+                :disabled="notAllowed"
+                >Bypass A Restriction</v-btn
+              >
+            </template>
+          </check-permission-role>
+        </template>
+      </body-header>
+      <content-group class="mt-4">
+        <toggle-content
+          :toggled="!bypassedRestrictions.length"
+          message="No bypassed restrictions found."
+        >
+          <DataTable
+            :value="bypassedRestrictions"
+            :paginator="true"
+            :rows="DEFAULT_PAGE_LIMIT"
+            :rowsPerPageOptions="PAGINATION_LIST"
+            :totalRecords="bypassedRestrictions.length"
+          >
+            <template #empty>
+              <p class="text-center font-weight-bold">No records found.</p>
+            </template>
+            <Column
+              field="restrictionType"
+              header="Restriction Type"
+              :sortable="true"
+            ></Column>
+            <Column field="restrictionCode" header="Restriction Code"></Column>
+            <Column header="View Details">
+              <template #body="{ data }">
+                <v-btn
+                  @click="$emit('viewBypassDetails', data.id)"
+                  color="primary"
+                  variant="text"
+                  class="text-decoration-underline"
+                  prepend-icon="fa:far fa-file-alt"
+                >
+                  View Details</v-btn
+                >
+              </template></Column
+            ><Column header="Remove Bypass Rule?">
+              <template #body="{ data }">
+                <v-btn
+                  @click="$emit('viewRemoveBypass', data.id)"
```

Comment:
May not be a part of this PR, but if the `emit` is provided and a placeholder is set, it has to be a conditional emit based on `isActive`. Please correct me if I am missing something.

Reply:
You're correct. `emit` is removed in this PR. It will be used based on `isActive` for the column "View Details", and I've also used another key "isBypassActive" for the column "Remove Bypass Rule?", which uses the same value for `isActive`. The reason to use another key is that Vuetify data table forbids duplicate keys, and I'm open for alternative suggestions.